src/open_inwoner/accounts/tests/test_auth.py

```diff
@@ -230,6 +231,48 @@ def test_user_can_modify_only_email_when_digid_and_brp(self, m):
         self.assertEqual(user.first_name, "Merel")
         self.assertEqual(user.last_name, "Kooyman")
 
+    def test_notification_settings_with_cms_page_published(self):
+        """
+        Assert that notification settings can be changed via the necessary-fields form
+        if the corresponding CMS pages are published. Fields corresponding to unpublished
+        pages should not be present.
+        """
+        cms_tools.create_apphook_page(
+            CollaborateApphook,
+            parent_page=self.homepage,
+        )
+
+        invite = InviteFactory()
+
+        url = reverse("digid-mock:password")
+        params = {
+            "acs": reverse("acs"),
+            "next": f"{reverse('profile:registration_necessary')}?invite={invite.key}",
+        }
+        url = f"{url}?{urlencode(params)}"
+
+        data = {
+            "auth_name": "533458225",
+            "auth_pass": "bar",
+        }
+
+        # post our password to the IDP
+        response = self.app.post(url, data).follow().follow()
+
+        necessary_form = response.forms[4]
```

Comment:
Using the form index is a bit risky as a modification elsewhere (eg: moving forms around) can mysteriously break this test. It is safer to use the name attribute so this test is independent from whatever is around the form we're interested in.